langchain/src/tools/gmail/create_draft.ts

```diff
@@ -0,0 +1,71 @@
+import { google } from "googleapis";
+import { encodeToBase64 } from "@gomomento/sdk-core/dist/src/internal/utils/string.js";
+import { GmailBaseTool, GmailBaseToolParams } from "./base.js";
+
+export interface CreateDraftSchema {
+  message: string;
+  to: string[];
+  subject: string;
+  cc?: string[];
+  bcc?: string[];
+}
+
+export class GmailCreateDraft
+  extends GmailBaseTool
+{
+
+  name = "create_gmail_draft";
+
+  description =
+    "Use this tool to create a draft email with the provided message fields.";
+
+  constructor(fields: GmailBaseToolParams) {
+    super(fields);
+  }
+
+  static args = [
+    { name: "message", description: "The message to include in the draft" },
+    { name: "to", description: "The list of recipients" },
+    { name: "subject", description: "The subject of the message" },
+    { name: "cc", description: "The list of CC recipients", isList: true },
+    { name: "bcc", description: "The list of BCC recipients", isList: true },
+  ];
+
+  protected prepareDraftMessage(message: string,
+    to: string[], subject: string, cc?: string[], bcc?: string[]) {
+        const draftMessage = {
+            message: {
+                raw: ""
+            }
+        }
+
+        const email = [
+            `To: ${to.join(", ")}`,
+            `Subject: ${subject}`,
+            cc ? `Cc: ${cc.join(", ")}` : "",
+            bcc ? `Bcc: ${bcc.join(", ")}` : "",
+            "",
+            message,
+          ].join("\n");
+
+          draftMessage.message.raw = encodeToBase64(email);
+
+          return draftMessage;
+    }
+
+    async _call(args: CreateDraftSchema) {
```

Comment:
doesnt this args param conflict with the static args field? is the static thing supposed to be a fallback if theres nothing passed in here or what? maybe they should have different names

Reply:
The python to JS translator turned ![image](https://github.com/1239uth/langchainjs/assets/90403097/782fccb0-04f5-4760-bca9-a8f6f0d837a9) into the static args array, not sure why this is useful in either repository. Should we remove it?

Reply:
Doesn't seem like we're using the static args anyways right? if so then yea lets remove it